executor/write_test.go

```diff
@@ -2169,7 +2169,6 @@ func (s *testSuite4) TestLoadData(c *C) {
 		{[]byte("\t2\t3"), []byte("\t4\t5"), nil, []byte("\t2\t3\t4\t5"), "Records: 0  Deleted: 0  Skipped: 0  Warnings: 0"},
 	}
 	checkCases(tests, ld, c, tk, ctx, selectSQL, deleteSQL)
-	c.Assert(sc.WarningCount(), Equals, uint16(1))
```

Comment:
This check is incorrect. The warnings is reset after the next statement.

The correct behavior should be check warning immediately after a statement.
The old code rely on every time a new statement context is allocated ... so even it does not checked immediately, the data is not overwritten.